Mathlib/MeasureTheory/Measure/Haar/Unique.lean

```diff
@@ -294,23 +295,40 @@ lemma haarScalarFactor_pos_of_isOpenPosMeasure (μ' μ : Measure G) [IsFiniteMea
   by_cases hG : LocallyCompactSpace G; swap
   · simp [haarScalarFactor, hG] at H
   -- Fix some nonzero continuous function with compact support `g`.
-  obtain ⟨g, g_cont, g_comp, g_nonneg, g_one⟩ :
-      ∃ (g : G → ℝ), Continuous g ∧ HasCompactSupport g ∧ 0 ≤ g ∧ g 1 ≠ 0 := by
-    rcases exists_compact_mem_nhds (1 : G) with ⟨k, hk, k_mem⟩
-    rcases exists_continuous_one_zero_of_isCompact hk isClosed_empty (disjoint_empty k)
-      with ⟨⟨g, g_cont⟩, gk, -, g_comp, hg⟩
-    refine ⟨g, g_cont, g_comp, fun x ↦ (hg x).1, ?_⟩
-    have := gk (mem_of_mem_nhds k_mem)
-    simp only [ContinuousMap.coe_mk, Pi.one_apply] at this
-    simp [this]
-  have int_g_pos : 0 < ∫ x, g x ∂μ' := by
-    apply (integral_pos_iff_support_of_nonneg g_nonneg _).2
-    · exact IsOpen.measure_pos μ' g_cont.isOpen_support ⟨1, g_one⟩
-    · exact g_cont.integrable_of_hasCompactSupport g_comp
+  obtain ⟨⟨g, g_cont⟩, g_comp, g_nonneg, g_one⟩ :
+      ∃ g : C(G, ℝ), HasCompactSupport g ∧ 0 ≤ g ∧ g 1 ≠ 0 := exists_continuous_nonneg_pos 1
+  have int_g_pos : 0 < ∫ x, g x ∂μ' := g_cont.integral_pos_of_hasCompactSupport_nonneg_nonzero
+    g_comp g_nonneg g_one
   have := integral_isMulLeftInvariant_eq_smul_of_hasCompactSupport μ' μ g_cont g_comp
   simp only [H, zero_smul, integral_zero_measure] at this
   linarith
 
+@[to_additive addHaarScalarFactor_of_eq_smul]
+lemma haarScalarFactor_of_eq_smul [LocallyCompactSpace G]
+    (μ' μ : Measure G) [IsFiniteMeasureOnCompacts μ] [IsFiniteMeasureOnCompacts μ']
+    [IsMulLeftInvariant μ] [IsMulLeftInvariant μ'] [IsOpenPosMeasure μ]
+    {c : ℝ≥0} (smul_eq : μ' = c • μ) : haarScalarFactor μ' μ = c := by
```

Comment:
Is there a reason not to write this as
```suggestion
    {c : ℝ≥0} : haarScalarFactor (c • μ) μ = c := by
```

Reply:
You're right :)
I changed the lemma to `haarScalarFactor (c • μ') μ = c • haarScalarFactor μ' μ` and 
`haarScalarFactor μ μ = 1` is now proved in a different way. These are both `simp`ed so the original lemma shouldn't really exist...
@eric-wieser